### PR TITLE
fixed logging for SendHostPluginHeartbeat and SendImdsHeartbeat PeriodicOperation 

### PR DESCRIPTION
**Pull request by nagworld9**
## Description

Issue # BUG 8892775

Utilized logging of periodic_operation base class for SendHostPluginHeartbeat and SendImdsHeartbeat if we can encounter exception and removed use of try/catch in MonitorHandler since we are already handling at periodic operation run
---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/ga/monitor.py
Comment (narrieta) [inline azurelinuxagent/ga/monitor.py:288]: the finally is needed; we still want to sleep until the next operation should execute

Comment (nagworld9) [reply azurelinuxagent/ga/monitor.py:288]: Is finally possible without try and my understanding is it never execute without try.

Comment (narrieta) [reply azurelinuxagent/ga/monitor.py:288]: you need try/finally (used to be try/except/finally)

**Review by larohra — APPROVED** [review]
Approving now but please dont merge without verifying the full end-to-end workflow

File: azurelinuxagent/ga/monitor.py
Comment (nagworld9) [reply azurelinuxagent/ga/monitor.py:288]: Addressed

**Review by dhivyaganesan — APPROVED** [review]

**Pull request merged**
